CMakeLists.txt

```diff
@@ -128,7 +128,12 @@ endif()
 
 ##################################################################################
 
-option(OCS_AVAILABLE "Online compiler support available to build pocl with. Default is available." ON)
+if(DEFINED OCS_AVAILABLE)
+  message(WARNING "The OCS_AVAILABLE option is deprecated, since it actually meant 'LLVM available', but LLVM is not the only way to get compiler support in a device. Please use ENABLE_LLVM if you want to enable/disable building pocl against LLVM.")
+  set(ENABLE_LLVM ${OCS_AVAILABLE} CACHE BOOL "build against LLVM" FORCE)
+else()
+  option(ENABLE_LLVM "Online compiler support available to build pocl with. Default is available." ON)
```

Comment:
I have missed that. I'll change the comment to something like "Enable LLVM support".

> we should have that per device driver

That's what i intend to do.

> Looking at what the macro does is it builds pocl in a mode where only offline compiled programs are supported

Only if you assume LLVM is the only way of compilation. Which it is - right now. I'll paste the commit message here, with emphasis added:

>    "OCS_AVAILABLE" was really used in the sense "is pocl compiled with LLVM",
>    **because the two were essentially the same** in past. However if we want to
>    **support drivers which don't use LLVM** for compilation, the two are not the
>    same anymore, and we need to handle cases where we have source compiler
>    support without LLVM.

The point of this commit is to start preparation for drivers which compile by other means than LLVM. Since OCS_AVAILABLE is used in the LLVM_AVAILABLE sense today, the most appropriate  way seemed to rename it, and then remove/replace where required (which will be in next PR).
POCL_ONLY_OFFLINE_COMPILED_PROGRAMS has an entirely different meaning.

Reply:
> Only if you assume LLVM is the only way of compilation.

True.